sdk/core/azure-core/src/http/bearer_token_authentication_policy.cpp

```diff
@@ -61,21 +59,49 @@ bool BearerTokenAuthenticationPolicy::AuthorizeRequestOnChallenge(
   return false;
 }
 
+namespace {
+bool TokenNeedsRefresh(
+    Azure::Core::Credentials::AccessToken const& cachedToken,
+    Azure::Core::Credentials::TokenRequestContext const& cachedTokenRequestContext,
+    Azure::DateTime const& currentTime,
+    Azure::Core::Credentials::TokenRequestContext const& newTokenRequestContext)
+{
+  return newTokenRequestContext.TenantId != cachedTokenRequestContext.TenantId
+      || newTokenRequestContext.Scopes != cachedTokenRequestContext.Scopes
+      || currentTime > (cachedToken.ExpiresOn - newTokenRequestContext.MinimumExpiration);
+}
+
+void ApplyBearerToken(
+    Azure::Core::Http::Request& request,
+    Azure::Core::Credentials::AccessToken const& token)
+{
+  request.SetHeader("authorization", "Bearer " + token.Token);
+}
+} // namespace
+
 void BearerTokenAuthenticationPolicy::AuthenticateAndAuthorizeRequest(
     Request& request,
     Credentials::TokenRequestContext const& tokenRequestContext,
     Context const& context) const
 {
-  std::lock_guard<std::mutex> lock(m_accessTokenMutex);
+  DateTime const currentTime = std::chrono::system_clock::now();
```

Comment:
Does this add the 1970 year bias to the `currentTime` variable? Looking at the std::chrono::time_point ctor, it doesn't appear to.

Reply:
`Azure::DateTime` is constructible (correctly constructible) from `std::chrono::system_clock::time_point`: it adds `SystemClockEpoch`:
https://github.com/Azure/azure-sdk-for-cpp/blob/b54d509c72fb76598832bcf1fcd5c4a89b245782/sdk/core/azure-core/inc/azure/core/datetime.hpp#L129